### PR TITLE
feat: advanced settings for custom feeds

### PR DESCRIPTION
**Pull request by capJavert**

### CODE REVIEW

File: src/schema/feeds.ts [hunk 1]
```diff
@@ -2233,18 +2238,20 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
...
-      const feed = await feedRepo.findOneBy({ id: feedId });
+      await feedRepo.findOneByOrFail({
+        id: feedId,
+        userId: ctx.userId,
+      });
       const feedAdvSettingsrepo = ctx.con.getRepository(FeedAdvancedSettings);
 
-      if (!feed) {
-        await feedRepo.save({ userId: feedId, id: feedId });
-      }
```
Comment (capJavert) [inline src/schema/feeds.ts:2246]: users have main feed by default so this fallback is no longer needed

File: src/schema/feeds.ts [hunk 2]
```diff
@@ -2233,18 +2238,20 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
...
       ctx: AuthContext,
     ): Promise<GQLFeedAdvancedSettings[]> => {
-      const feedId = ctx.userId;
+      const feedId = feedIdArg || ctx.userId;
       const feedRepo = ctx.con.getRepository(Feed);
-      const feed = await feedRepo.findOneBy({ id: feedId });
+      await feedRepo.findOneByOrFail({
+        id: feedId,
+        userId: ctx.userId,
+      });
```
Comment (capJavert) [inline src/schema/feeds.ts:2252]: only update settings for user owned feeds

File: __tests__/feeds.ts [hunk 3]
```diff
@@ -2147,6 +2151,63 @@ describe('mutation updateFeedAdvancedSettings', () => {
     });
     expect(res.data).toMatchSnapshot();
   });
+
+  it('should update custom feed advanced settings', async () => {
+    loggedUser = '1';
+
```
Comment (rebelchris) [inline __tests__/feeds.ts:2157]: We should add @feedPlus here right, then this will fail without setting isPlus

File: src/schema/feeds.ts [hunk 4]
```diff
@@ -894,6 +894,11 @@ export const typeDefs = /* GraphQL */ `
     Update user's feed advanced settings
     """
     updateFeedAdvancedSettings(
```
Comment (rebelchris) [inline src/schema/feeds.ts:896]: We need to add feedPlus here.

**Review by rebelchris — CHANGES_REQUESTED** [review]
We need feedPlus right?

File: src/schema/feeds.ts (hunk 4 above)
Comment (capJavert) [reply src/schema/feeds.ts:896]: yes, gonna update after your PR is merged

**Review by rebelchris — APPROVED** [review]

**Pull request merged**